test/Constraints/rdar82828226.swift

```diff
@@ -0,0 +1,8 @@
+// RUN: %target-typecheck-verify-swift
+
+import Foundation
+
+func f(_ n: NSNumber) {}
+let int: Int = 0
+f(int) // expected-error {{cannot convert value of type 'Int' to expected argument type 'NSNumber'}} {{3-3=NSNumber(value: }}
```

Comment:
Just an idea @xedin, but since `Int` can be bridged to `NSNumber` we could instead provide an bridging coercion fix-it e.g. `f(int as NSNumber)`

Reply:
I think that might be preferable, only caveat - we need to figure out where `as NSNumber` needs to be wrapped into parens - `exprNeedsParensBeforeAddingAs` could be used for that. Another reason why this suggestion might be better than initializer - it allows us to support NSInteger and other types.